meshnets/utils/data_loading.py

```diff
@@ -1,17 +1,107 @@
-"""Load the mesh data"""
+""""Methods for loading mesh data"""
 
+from typing import Union
 from typing import Tuple
 
 from absl import logging
 import meshio
 import numpy as np
+import pyvista as pv
 
-# TODO(victor): modify the loader once a data format has been decided
 
+def load_edge_mesh_pv(
+    obj_path: str,
+    get_pressure: bool = True,
+    verbose: bool = False
+) -> Union[Tuple[np.ndarray, np.ndarray, np.ndarray], Tuple[np.ndarray,
+                                                            np.ndarray]]:
+    """Extract node coordinates, edges and pressure at nodes from a mesh file
+    using pyvista."""
 
-def load_mesh_from_obj(obj_path: str,
+    mesh = pv.read(obj_path)
+    # `use_all_points=False` discards duplicates and renumber the nodes
+    # TODO(victor): verify that node pressure array is modified accordingly
+    # WARNING: the remeshing should be done prior on the mesh file, not here
+    edge_mesh = mesh.extract_all_edges(use_all_points=True)
+
+    nodes = edge_mesh.points
+    edge_list = edge_mesh.lines.reshape((-1, 3))[:, 1:]
+
+    if verbose:
+        logging.info('Nodes shape : %s', nodes.shape)
+        logging.info('Edge list shape : %s', edge_list.shape)
+
+    if get_pressure:
+        # TODO(victor): decide if array reshape is necessary
+        pressure = edge_mesh.get_array('p', preference='point').reshape(-1, 1)
+        if verbose:
+            logging.info('Pressure shape : %s', pressure.shape)
+
+        return nodes, edge_list, pressure
+    else:
+        return nodes, edge_list
+
+
+def load_edge_mesh_meshio(
+    obj_path: str,
+    get_pressure: bool = True,
+    verbose: bool = False
+) -> Union[Tuple[np.ndarray, np.ndarray, np.ndarray], Tuple[np.ndarray,
+                                                            np.ndarray]]:
+    """Extract node coordinates, edges and pressure at nodes from a mesh file
+    using meshio.
+    
+    Edges can include duplicates.
+    """
+
+    mesh = meshio.read(obj_path)
+    # `byteswap().newbyteorder('<')` is required because the mesh points array
+    # is sometimes in big endian byte order, which is not supported by torch
+    # Behavior observed on .vtk files
+    # TODO: this should probably be avoided at file level
+    nodes = mesh.points.byteswap().newbyteorder('<')
+
+    # extract all the edges in the mesh
+    edges_list = []
```

Comment:
Create another separate function that implements this functionality from a mesh.